kediatr-core/src/test/kotlin/com/trendyol/CommandHandlerTest.kt

```diff
@@ -3,53 +3,61 @@ package com.trendyol
 import com.trendyol.kediatr.*
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.runBlocking
+import org.junit.Test
 import org.junit.jupiter.api.Nested
-import org.junit.jupiter.api.Test
 import kotlin.test.assertEquals
 import kotlin.test.assertFailsWith
 import kotlin.test.assertNotNull
-import kotlin.test.assertTrue
-
-private var counter = 0
-private var asyncTestCounter = 0
 
 class CommandHandlerTest {
 
-    init {
-        counter = 0
-        asyncTestCounter = 0
-    }
-
     @Test
-    fun `commandHandler should be fired`() {
-        val handler = MyCommandHandler()
-        val handlers: HashMap<Class<*>, Any> = hashMapOf(Pair(MyCommandHandler::class.java, handler))
-        val provider = ManuelDependencyProvider(handlers)
-        val bus: CommandBus = CommandBusBuilder(provider).build()
-        bus.executeCommand(MyCommand())
+    fun commandHandler_should_be_fired() {
+        class TestCommand : Command
 
-        assertTrue {
-            counter == 1
+        class TestCommandHandler : CommandHandler<TestCommand> {
+            var invocationCount = 0
+            override fun handle(command: TestCommand) {
+                invocationCount++
+            }
         }
+
+        val handler = TestCommandHandler()
+        val handlers: HashMap<Class<*>, Any> = hashMapOf(Pair(TestCommandHandler::class.java, handler))
+        val provider = ManualDependencyProvider(handlers)
+        val bus: CommandBus = CommandBusBuilder(provider).build()
+        bus.executeCommand(TestCommand())
+
+        assertEquals(1, handler.invocationCount)
     }
 
     @Test
-    fun `async commandHandler should be fired`() = runBlocking {
+    fun async_commandHandler_should_be_fired() = runBlocking {
```

Comment:
[runTest](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/run-test.html) can be used here instead of runBlocking, runTest allows using delay without causing the tests to take more time than necessary.

Reply:
Actually, there is no need to use `Delay` in the tests; all of them can be scoped for the test itself. I will change them while deleting the sync code.

I removed it now.